weather.php

```diff
@@ -1,30 +1,33 @@
 <?php
-  /**
-   * This page talks to weatherapi.com.
-   */
 
-  // API URL.
-  // You can register for free to get your own api key (fb0... below). 
-  $url = "http://api.weatherapi.com/v1/current.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&q=";
+/**
+ * This page talks to weatherapi.com.
+ */
 
-  // Check and filter query parameters.
-  $query = filter_input(INPUT_GET, 'query', FILTER_DEFAULT);
+// API URL.
+// You can register for free to get your own api key (fb0... below). 
+$currentUrl = "http://api.weatherapi.com/v1/current.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&q=";
+$forecastUrl = "http://api.weatherapi.com/v1/forecast.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&days=3&q=";
 
-  if ($query) {
-    $query = urlencode($query);
+// Check and filter query parameters.
+$query = filter_input(INPUT_GET, 'query', FILTER_DEFAULT);
 
-    // Call the api with the user input query (zip, city, etc.).
-    $data = file_get_contents($url . $query);
-    $parsed = json_decode($data);
+if ($query) {
+  $query = urlencode($query);
 
-    // You can use PHP's error_log to output a string to the log for debugging.
-    // error_log($parsed->location->name);
+  // Call the api with the user input query (zip, city, etc.).
+  $data = file_get_contents($forecastUrl . $query);
+  $parsed = json_decode($data);
 
-    $result['weather'] = $parsed->current->condition->text;
-    $result['temp_f'] = $parsed->current->temp_f;
-    $result['icon_url'] = $parsed->current->condition->icon;
-    $result['city'] = $parsed->location->name;
-    $result['state'] = $parsed->location->region;
-    header('Content-Type: application/json');
-    echo json_encode($result);
-  }
+  // You can use PHP's error_log to output a string to the log for debugging.
+  // error_log($parsed->location->name);
+
+  $result['weather'] = $parsed->current->condition->text;
+  $result['temp_f'] = $parsed->current->temp_f;
+  $result['icon_url'] = $parsed->current->condition->icon;
+  $result['city'] = $parsed->location->name;
+  $result['state'] = $parsed->location->region;
+  $result['forecast'] = $parsed->forecast->forecastday;
+  header('Content-Type: application/json');
+  echo json_encode($result);
+}
```

Comment:
Maybe you could have an else here to return json with an error like "You need to enter a valid location."

Reply:
Done, added two validation types: Invalid input (int <= 99999) and Invalid Location